include/clover_field.h

```diff
@@ -10,6 +10,19 @@ namespace quda {
   namespace clover
   {
 
+    /**
+       @brief Helper function that returns whether we have enabled
+       clover fermions.
+     */
+    constexpr bool is_enabled()
```

Comment:
This isn't a hill I'll die on, but I think I'd personally prefer if this was called `is_clover_enabled()`, even though I see it's in the `clover` namespace, to absolutely minimize potential ambiguity.

This would be consistent with precedent set in `instatiate.h`, where the enabled functions either include a specifier in the function name or require an explicit template parameter that serves the same self-clarifying purpose.

Reply:
I had a different design in mind, the `clover::is_enabled()` was one of expediency.  I might as well just push the real one....
``` c++
constexpr bool is_enabled(QudaDiracType);
```
Will add this in my next round of pushes.

Reply:
After hacking around for a while, and realizing that a global edit for deploying `is_enabled()` throughout for Dirac type enablement is not appropriate for this PR, I've gone with a narrow fix.

6cefd1c4816e9c3750ad89126450e59a887620ee